cli/src/lib.rs

```diff
@@ -40,22 +40,23 @@ enum PoolFormat {
 
 impl Display for PoolFormat {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let s = format!("{:?}", self);
+        let s = format!("{self:?}");
         write!(f, "{}", s.to_lowercase())
     }
 }
 
 fn parse_pool_path(pool_path: &str) -> (&Path, crate::PoolFormat) {
     let path = Path::new(pool_path);
-    let format = match path.extension() {
-        Some(ext) => match ext.to_str().unwrap() {
-            "json" => PoolFormat::Json,
-            "yaml" | "yml" => PoolFormat::Yaml,
-            _ => {
-                eprintln!("Unknown pool file extension: {}", ext.to_str().unwrap());
-                exit(exitcode::USAGE);
-            }
-        },
+    let format = match path
+        .extension()
+        .map(|v| v.to_str().expect("str because path created from str"))
```

Comment:
with the rebase, I wanted to double-check that you saw I reverted to using `to_str`

Reply:
ok than kyou